tools/toolsrc/pack_tool/main.c

```diff
@@ -17,8 +17,8 @@ typedef struct{
 
 unsigned int HashGen(unsigned char* file, unsigned int size){  //that's the simplest and crappiest hash engine i could invent
 	unsigned int HASH = 0;							  //but it works, so i don't give a shit
-	int i;
-	for(i = 0; i < size - 4; i++){
+	unsigned int i;
+	for(i = 0; i + 4 < size; i++){
```

Comment:
Right if the size less than 4, original code might *BOOM* (kidding).
How did you find it this time?